src/main.c

```diff
@@ -19,6 +19,18 @@
 
 #define _cleanup_(x) __attribute__((cleanup(x)))
 
+#ifndef _SD_DEFINE_POINTER_CLEANUP_FUNC
```

Comment:
The macro comes from libsystemd's `_sd-common.h` private header.
I'd add a comment with that path: a) gives credit, b) provides hints where to look at if systemd changes it in the future.

Reply:
Got it.